packages/salesforce-adapter/src/filters/convert_maps.ts

```diff
@@ -264,30 +264,37 @@ const convertArraysToMaps = (
       .filter(([fieldName]) => _.get(getElementValueOrAnnotations(element), fieldName) !== undefined)
       .forEach(([fieldName, mapDef]) => {
         const mapper = mapDef.mapper ?? defaultMapper
+        const mapDefFilter = (item: Values): boolean => typeof item[mapDef.key] === 'string'
```

Comment:
```suggestion
        const isConvertableValue = (item: Values): boolean => _.isString(item[mapDef.key])
```
Also, to be extra safe I would implement the following:
const isConvertableValue = (item: unknown): boolean =>_.isPlainObject(item) &&  _.isString(item[mapDef.key])

Reply:
Done.